### DISCUSSION TITLE
Bug: github.job is empty in jobs.if

### DISCUSSION THREAD

**Discussion by mvlasovatl** [community/community#25660] [Actions]
<p><strong>Test workflow:</strong></p>
<pre><code class="lang-yaml">on: push
jobs:
  runs:
    runs-on: ubuntu-latest
    if: github.job == ''
    steps:
      - run: echo Github current job [${{ github.job }}]
</code></pre>
<p><strong>Expected:</strong> job is failed.<br>
<strong>Actual:</strong> job is passing.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2bb5f9ff4c0f811653b0af8bf675c101eb0ec3f1.png" data-download-href="/uploads/short-url/6eGpG9SlclSAdzUWwGbCVxYaXHb.png?dl=1" title="GhActionsBug" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/66078253/181092405-76690eb3-185c-46a3-b6ea-7d8708b27141.png" alt="GhActionsBug" data-base62-sha1="6eGpG9SlclSAdzUWwGbCVxYaXHb" width="690" height="346" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/2bb5f9ff4c0f811653b0af8bf675c101eb0ec3f1_2_690x346.png, https://aws1.discourse-cdn.com/github/original/2X/2/2bb5f9ff4c0f811653b0af8bf675c101eb0ec3f1.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/2/2bb5f9ff4c0f811653b0af8bf675c101eb0ec3f1.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2bb5f9ff4c0f811653b0af8bf675c101eb0ec3f1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">GhActionsBug</span><span class="informations">722×363 26.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Please fix it. I wanted to use <code>github.job</code> together with <code>workflow_dispatch</code> to run custom jobs:</p>
<pre><code class="lang-yaml">if: github.event.inputs.jobs == '' || contains(github.event.inputs.jobs, github.job)
</code></pre>

**Comment by NicoNBW** [comment]
<p>Thanks for your feedback.<br>
Before the job runs, the value of github.job is always empty. Only when the job is actually running, we can get the job context. If you are using this if conditional in jobs level, the value of github.job is always empty. So no matter what, your job will not be skipped. This is why you can not use if conditional at jobs level. Please use the if conditional in steps level. For example:</p>
<pre><code class="lang-auto">on:
  workflow_dispatch:
    inputs:
      jobs:

jobs:
  build:
    runs-on: ubuntu-latest
        
    steps:
      - name: test 
        if: ${{ github.event.inputs.jobs == '' || contains(github.event.inputs.jobs, github.job)}}
        run: echo Github current job [${{ github.job }}]

</code></pre>

**Comment by mvlasovatl** [comment, marked as answer]
<p>Yeah, that might work as well. In my case though the job contains up to 10 steps, so it would be inconvenient to paste that boilerplate code to every step. I managed to achieve the logic I wanted by hard-coding of the job ID in <code>if</code> clause:</p>
<pre><code class="lang-yaml">on:
  workflow_dispatch:
    inputs:
      jobs:
        description: Jobs to run (defaults to all)
        required: false
jobs:    
  unit-tests:
    runs-on: ubuntu-latest
    if: github.event.inputs.jobs == '' || contains(github.event.inputs.jobs, 'unit-tests')
</code></pre>

**Comment by jansiegel** [comment]
<aside class="quote no-group" data-username="niconbw" data-post="2" data-topic="122038">
<div class="title">
<div class="quote-controls"></div>
 niconbw:</div>
<blockquote>
<p>This is why you can not use if conditional at jobs level.</p>
</blockquote>
</aside>
<p>But it’s supposed to be supported, right? This problem seems to still be occurring, and I cannot find anything about the impossibility of using <code>github.jobs</code> in the jobs’ <code>if</code> context.</p>
